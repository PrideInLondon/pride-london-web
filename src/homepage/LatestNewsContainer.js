import React from 'react'
import { StaticQuery, graphql } from 'gatsby'
import { Row, FlexColumn } from '../components/grid'
import constants from '../constants'
import { CTALink } from '../components/ctaLink'
import LatestNewsHeader from './LatestNewsHeader'
import LatestNews from './LatestNews'
import {
  Wrapper,
  SecondaryNews,
  NewsCard,
  CTALinkOnlyMobile,
} from './LatestNewsContainer.styles'

export const query = graphql`
  query latestNewsQuery {
    articles: allContentfulArticle(
      filter: { category: { ne: "Views" } }
      sort: { fields: [datePublished], order: DESC }
      limit: 3
    ) {
      edges {
        node {
          id
          slug
          datePublished
          title
          category
          headerImage {
            title
            fixed(
              width: 830
              height: 500
              resizingBehavior: FILL
              quality: 90
            ) {
              ...GatsbyContentfulFixed
            }
          }
        }
      }
    }
  }
`

const LatestNewsContainer = () => (
  <Wrapper>
    <Row>
      <FlexColumn width={1}>
        <LatestNewsHeader />
      </FlexColumn>
    </Row>

    <Row mx={[0, 0, 0, 75]}>
      <StaticQuery
        query={query}
        render={({ articles: { edges: newsArr = [] } = {} }) => (
          <>
            <FlexColumn
              width={[
                1, // 100% between 0px screen width and first breakpoint (375px)
                1, // 100% between first breakpoint(375px) and second breakpoint (768px)
                1, // 100% between second breakpoint(768px) and third breakpoint (1024px)
                2 / 3, // 66% between third breakpoint(1280px) and fourth breakpoint (1440px)
              ]}
              px={[0, 0, 0, 15]}
            >
              <LatestNews
                {...{
                  ...newsArr[0].node,
                  category: constants.articleCategories.find(
                    cat => cat.title == newsArr[0].node.category
                  ),
                }}
              />
            </FlexColumn>
            <FlexColumn
              width={[
                1, // 100% between 0px screen width and first breakpoint (375px)
                1, // 100% between first breakpoint(375px) and second breakpoint (768px)
                1, // 100% between second breakpoint(768px) and third breakpoint (1024px)
                1 / 3, // 33% between third breakpoint(1280px) and fourth breakpoint (1440px)
              ]}
            >
              <SecondaryNews>
                {newsArr.slice(1).map(({ node: singleNews }) => (
                  <NewsCard
                    key={singleNews.id}
                    {...{
                      ...singleNews,
                      category: constants.articleCategories.find(
                        cat => cat.title == singleNews.category
                      ),
                    }}
                  />
                ))}
              </SecondaryNews>
            </FlexColumn>
          </>
        )}
      />
    </Row>
    <Row>
      <FlexColumn
        width={[
          1, // 100% between 0px screen width and first breakpoint (375px)
          1, // 100% between first breakpoint(375px) and second breakpoint (768px)
          1, // 100% between second breakpoint(768px) and third breakpoint (1024px)
          1 / 3, // 33% between third breakpoint(1280px) and fourth breakpoint (1440px)
        ]}
        py={[1, 2, 0]}
      >
        <CTALinkOnlyMobile>
          <CTALink to="/news-and-views">View all news</CTALink>
        </CTALinkOnlyMobile>
      </FlexColumn>
    </Row>
  </Wrapper>
)

export default LatestNewsContainer
