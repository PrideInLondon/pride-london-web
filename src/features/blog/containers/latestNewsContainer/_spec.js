import React from 'react'
import { shallow } from 'enzyme'
import LatestNewsContainer from '.'

describe(LatestNewsContainer.name, () => {
  it('renders with default values', () => {
    const wrapper = shallow(<LatestNewsContainer />)
    expect(wrapper).toMatchSnapshot()
  })
})
