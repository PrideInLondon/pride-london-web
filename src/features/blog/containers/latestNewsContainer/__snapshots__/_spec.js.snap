// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LatestNewsContainer renders with default values 1`] = `
<styles__Container>
  <Row>
    <grid__FlexColumn
      width={
        Array [
          1,
          1,
          1,
          1,
        ]
      }
    >
      <LatestNewsHeader />
    </grid__FlexColumn>
  </Row>
  <Row>
    <StaticQuery
      data={
        Object {
          "data": Object {
            "allContentfulNews": Object {
              "edges": Array [
                Object {
                  "node": Object {
                    "date": "2019-04-01T20:30+00:00",
                    "id": "12f80ebd-6465-5042-9b26-a07e5946b0e7",
                    "images": Array [
                      Object {
                        "file": Object {
                          "url": "//images.ctfassets.net/n2o4hgsv6wcx/5Dt5BK1ydaGKkW6s60kii2/99ce4f6dbf5f3242212c39112ea2c94e/monty-python-black-knight-2.png",
                        },
                      },
                    ],
                    "newsCategory": Object {
                      "hexColour": "#ed2c6e",
                      "title": "Research",
                    },
                    "title": "This is still a featured article 2 with Image",
                  },
                },
                Object {
                  "node": Object {
                    "date": "2019-04-01T20:25+00:00",
                    "id": "d67f467d-f2f0-5039-84d4-0068b6bf35e3",
                    "images": Array [
                      Object {
                        "file": Object {
                          "url": "//images.ctfassets.net/n2o4hgsv6wcx/5Dt5BK1ydaGKkW6s60kii2/99ce4f6dbf5f3242212c39112ea2c94e/monty-python-black-knight-2.png",
                        },
                      },
                    ],
                    "newsCategory": Object {
                      "hexColour": "#ed2c6e",
                      "title": "Research",
                    },
                    "title": "This is still a featured article 3 with Image",
                  },
                },
                Object {
                  "node": Object {
                    "date": "2019-04-01T20:20+00:00",
                    "id": "e4c062b9-29ac-5e8a-882d-6c47c0b3f90c",
                    "images": Array [
                      Object {
                        "file": Object {
                          "url": "//images.ctfassets.net/n2o4hgsv6wcx/5Dt5BK1ydaGKkW6s60kii2/99ce4f6dbf5f3242212c39112ea2c94e/monty-python-black-knight-2.png",
                        },
                      },
                    ],
                    "newsCategory": Object {
                      "hexColour": "#ed2c6e",
                      "title": "Research",
                    },
                    "title": "This is still a featured article 1 with Image",
                  },
                },
              ],
            },
          },
        }
      }
      query="2270872770"
      render={[Function]}
    />
  </Row>
</styles__Container>
`;
